### PR TITLE
FIX: Fixes #8676 Added validation for constructor input

### LINKED ISSUE #8676: DBField constructor input validation
## Affected Version

SilverStripe 4.x

## Description

`DBField` constructor does not validate input properly which may potentially lead to bugs in resulting SQL queries.

## Steps to Reproduce

```php
new \SilverStripe\ORM\FieldType\DBVarchar('mychar"');
```

# Related PRs
* https://github.com/silverstripe/silverstripe-framework/pull/8737

### PR DESCRIPTION
**Pull request by LukeAmos**
Fixes #8676

### CODE REVIEW

File: src/ORM/FieldType/DBField.php [hunk 1]
```diff
@@ -136,6 +136,10 @@ abstract class DBField extends ViewableData implements DBIndexable
      */
     public function __construct($name = null, $options = [])
     {
+        if (preg_match("#[^\w]#", $name)) {
```
Comment (maxime-rainville) [inline src/ORM/FieldType/DBField.php]: I think this expression is a bit too restrictive. It's relatively common to define table field names like "field-with-dashes" or "field_with_underscores". While it's definitely not a common set up in the SilverStripe ORM, I don't think we should preclude it either.

**Comment by maxime-rainville** [comment]
@LukeAmos Do you think you'll get another change to look at this one?

**Comment by LukeAmos** [comment]
@maxime-rainville sorry I haven't got back to it yet, should be able to in a couple of weeks.

**Comment by emteknetnz** [comment]
This pull request hasn't had any activity for a while. Are you going to be doing further work on it, or would you prefer to close it now?

**Comment by emteknetnz** [comment]
It seems like there's going to be no further activity on this pull request, so we've closed it for now. Please open a new pull-request if you want to re-approach this work, or for anything else you think could be fixed or improved.

**Pull request closed without merging**